sympy/polys/densetools.py

```diff
@@ -1199,13 +1199,20 @@ def dup_clear_denoms(f, K0, K1=None, convert=False):
     for c in f:
         common = K1.lcm(common, K0.denom(c))
 
-    if not K1.is_one(common):
-        f = dup_mul_ground(f, common, K0)
+    if K1.is_one(common):
+        if not convert:
+            return common, f
+        else:
+            return common, dup_convert(f, K0, K1)
```

Comment:
`return common, f if not convert else return common, dup_convert(f, K0, K1)`

Make the code smaller and also easier to read. Same for lines 1212-1215

Reply:
okay sir

Reply:
Making the code all on one line does not make it easier to read

Reply:
I also share the opinion that one-liner syntax is not logically shorter, and look esoteric in some cases like this.

If there is problem about the code about if-statement,
I think that the bigger problem is that the code just have if-statements. 
I would rather avoid writing any function, if I am going to write again. branches like `convert`, for instance, that looks very trivial, and can just be postprocessed.